src/Raven.Server/Documents/DatabasesLandlord.cs

```diff
@@ -1623,14 +1623,15 @@ public sealed class StateChange
 
             public long LastIndexChange;
 
-            public StateChange(ServerStore serverStore, string name, Logger logger, Action<DatabaseRecord, long> onChange, long lastIndexChange, CancellationToken token)
+            public StateChange(ServerStore serverStore, string name, Logger logger, Action<DatabaseRecord, long> onChange, long lastIndexChange, CancellationToken token, SemaphoreSlim locker = null)
             {
                 ServerStore = serverStore;
                 Name = name;
                 Logger = logger;
                 OnChange = onChange;
                 LastIndexChange = lastIndexChange;
                 Token = token;
+                Locker = locker ?? new SemaphoreSlim(1, 1);
```

Comment:
How often we will create this? Is this disposable? IIRC, yes? Do we need to handle that?

Reply:
On each creation of `DocumentDatabase`.
For sharding on creation of `ShardedDatabaseContext` for a database.